### PR TITLE
[turns][sound] Using for turn notification a real distance to the closest turn instead of a distance based on speed.

### PR DESCRIPTION
**Pull request by bykoianko**
При проигрывании поворота правильно использовать расстояние оставшееся до поворота, а не расстояние, которое вычисляется из скорости и времени активации оповещения.
Бывают случаи (после перестроения маршрута), когда эти расстояния могут сильно различаться.

PR помечен DO NOT MERGE, поскольку я жду, когда в мастер попадет https://github.com/mapsme/omim/pull/164 и я смогу доделать юнит тесты без конфликтов.

https://trello.com/c/xTfj3uHV/1898--

### CODE REVIEW

**Comment by mapsmetest** [comment]
Jenkins asks: can anyone have a look at this request?
If you want Jenkins to compile and test this, write "test this please" or JT_TP (without the underscore in the middle).

File: routing/turns_sound_settings.cpp [hunk 1]
```diff
@@ -73,7 +72,23 @@ double Settings::ConvertUnitsToMeters(double distanceInUnits) const
...
-  // m_lengthUnits is equal to LengthUnits::Undefined or to unknown value.
+  ASSERT(false, ());
+  return 0.;
+}
+
+double Settings::ConvertMetersToUnits(double distanceInMeters) const
+{
+  switch (m_lengthUnits)
+  {
+    case LengthUnits::Undefined:
```
Comment (deathbaba) [inline routing/turns_sound_settings.cpp]: Do we really need this Undefined state in the code? If it is not used, it's better to stick to some default value (meters?) or always get it from our settings::

**Comment by deathbaba** [comment]
LGTM with minor comment.

File: routing/turns_sound_settings.cpp (hunk 1 above)
Comment (bykoianko) [inline routing/turns_sound_settings.cpp]: Yes, you're right. I needed Undefined before but not now. 

I guess we need:
1. to remove Undefined form LengthUnits;
2. to move LengthUnits to platform/settings.hpp and to remove Unit enum there;
3. Use enum class LengthUnits instead of enum Unit;

But I think it's a question of separate PR.

Comment (deathbaba) [inline routing/turns_sound_settings.cpp]: Is it hard to refactor here?
Is it the only place where Undefined is used?

Comment (bykoianko) [inline routing/turns_sound_settings.cpp]: > Is it hard to refactor here?

I guess it'll be a PR with one commit of about 100 lines(addition/deletion). 

> Is it the only place where Undefined is used?

No. Of cause not. There are about 10 places where LengthUnits::Undefined is used. They shall be removed. 
Besides that there are some lines which shall be exchanged. Units::Metric -> LengthUnits::Meters and so on.

I guess we should do it but in a separate PR.

File: routing/routing_tests/turns_sound_test.cpp [hunk 2]
```diff
@@ -203,9 +203,13 @@ UNIT_TEST(TurnsSoundMetersTwoTurnsTest)
   TEST(turnNotifications.empty(), ());
 
   // 700 meters till the turn. It's time to pronounce the first voice notification.
```
Comment (kshalnev) [inline routing/routing_tests/turns_sound_test.cpp]: 600 meters till the turn, I guess.

**Comment by kshalnev** [comment]
LGTM
